storage/src/adb/benches/current_init.rs

```diff
@@ -0,0 +1,171 @@
+use commonware_cryptography::{hash, Hasher, Sha256};
+use commonware_runtime::{
+    benchmarks::{context, tokio},
+    buffer::PoolRef,
+    create_pool,
+    tokio::{Config, Context, Runner},
+    Runner as _, ThreadPool,
+};
+use commonware_storage::{
+    adb::current::{Config as CConfig, Current},
+    translator::EightCap,
+};
+use criterion::{criterion_group, Criterion};
+use rand::{rngs::StdRng, RngCore, SeedableRng};
+use std::time::Instant;
+use tracing::info;
+
+const NUM_ELEMENTS: u64 = 100_000;
+const NUM_OPERATIONS: u64 = 1_000_000;
+const COMMIT_FREQUENCY: u32 = 10_000;
+const DELETE_FREQUENCY: u32 = 10; // 1/10th of the updates will be deletes.
+const ITEMS_PER_BLOB: u64 = 500_000;
+const PARTITION_SUFFIX: &str = "current_bench_partition";
+
+/// Use a "prod sized" page size to test the performance of the journal.
+const PAGE_SIZE: usize = 16384;
+
+/// The number of pages to cache in the buffer pool.
+const PAGE_CACHE_SIZE: usize = 10_000;
+
+/// current_init is not singlethreaded. So we benchmark with 1 and 8 threads to compare singlethreaded and multithreaded performance.
+const SINGLE_THREADED: usize = 1;
+const MULTI_THREADED: usize = 8;
+
+/// Chunk size for the current ADB bitmap - must be a power of 2 (as assumed in current::grafting_height()) and a multiple of digest size (for proof size optimization).
+const CHUNK_SIZE: usize = 32;
+
+fn current_cfg(pool: ThreadPool) -> CConfig<EightCap> {
+    CConfig::<EightCap> {
+        mmr_journal_partition: format!("journal_{PARTITION_SUFFIX}"),
+        mmr_metadata_partition: format!("metadata_{PARTITION_SUFFIX}"),
+        mmr_items_per_blob: ITEMS_PER_BLOB,
+        mmr_write_buffer: 1024,
+        log_journal_partition: format!("log_journal_{PARTITION_SUFFIX}"),
+        log_items_per_blob: ITEMS_PER_BLOB,
+        log_write_buffer: 1024,
+        bitmap_metadata_partition: format!("bitmap_metadata_{PARTITION_SUFFIX}"),
+        translator: EightCap,
+        thread_pool: Some(pool),
```

Comment:
For the single-threaded case, you want to avoid using a threadpool entirely.  Perhaps make current_cfg accept Option<ThreadPool> instead, and then set it to None when threads == 1.